api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -201,83 +224,75 @@ public Builder setAttribute(String key, AttributeValue value) {
     }
 
     /**
-     * Sets a String {@link AttributeValue} into this.
+     * Sets a String attribute into this.
      *
      * @return this Builder
      */
     public Builder setAttribute(String key, String value) {
-      AttributeValue v = stringAttributeValue(value);
-      return setAttribute(key, v);
+      return setAttribute(stringKey(key), value);
     }
 
     /**
-     * Sets a long {@link AttributeValue} into this.
+     * Sets a long attribute into this.
      *
      * @return this Builder
      */
     public Builder setAttribute(String key, long value) {
```

Comment:
Should this change to LongKey, long value?

Reply:
My thought was, since these are already type-safe, to keep the easy option in the API, for cases where people don't want to directly use the new keys. The method that takes the generic key should cover the particular, and would end up being the same method signature as this if we change the first parameter to `LongKey`.

Reply:
Yeah I don't think we need to change these methods to use the keys. We could consider removing these methods - I'm on the fence on how useful these are for users. Keys will always have a significant performance benefit compared to creating a wrapper key every time this is invoked, so removing them reduces the chances of a user not realizing that.

Reply:
At the very least, I was thinking we should add some strongly worded language to the javadoc about preferring to use the method that takes the Key.

Reply:
added some javadoc to the relevant methods.